### PR TITLE
fix: goreleaser-build-sign-publish-chainlink workflow

### PR DESCRIPTION
**Pull request by momentmaker**
This refactors the `goreleaser-build-sign-publish-chainlink` workflow with:

- cosign using github OIDC
- fixes the actual build based on tag push
- general cleanup for goreleaser-build-sign-publish
- update the current docker image build to use keyless cosign as well

### for goreleaser

https://github.com/smartcontractkit/chainlink/actions/runs/10425492953

Verified with:

`cosign verify public.ecr.aws/chainlink/chainlink:v2.14.7-goreleaser-test8 \
      --certificate-oidc-issuer https://token.actions.githubusercontent.com \
      --certificate-identity "https://github.com/smartcontractkit/chainlink/.github/workflows/build-publish.yml@refs/tags/v2.14.7-goreleaser-test8"`

Output:

```
Verification for public.ecr.aws/chainlink/chainlink:v2.14.7-goreleaser-test8 --
The following checks were performed on each of these signatures:
  - The cosign claims were validated
  - Existence of the claims in the transparency log was verified offline
  - The code-signing certificate was verified using trusted certificate authority certificates

[{"critical":{"identity":{"docker-reference":"public.ecr.aws/chainlink/chainlink"},"image":{"docker-manifest-digest":"sha256:ff509c9118c14d2b45055c9c5b9ed203f931bb07e74ee0317f15c97d808de1cf"},"type":"cosign container image signature"},"optional":{"1.3.6.1.4.1.57264.1.1":"https://token.actions.githubusercontent.com","1.3.6.1.4.1.57264.1.2":"push","1.3.6.1.4.1.57264.1.3":"193f36b442b870b4c11c5aac5c6489cce19ac49e","1.3.6.1.4.1.57264.1.4":"Build, Sign and Publish Chainlink","1.3.6.1.4.1.57264.1.5":"smartcontractkit/chainlink","1.3.6.1.4.1.57264.1.6":"refs/tags/v2.14.7-goreleaser-test8","Bundle":{"SignedEntryTimestamp":"MEUCIQDmpW5MjLe+6SvM6m6qr0kRrPHaD5KjFqb9ZEJSNHWT3QIgZvx//D1lhaw0eoHOGVkeroDjXC9U8uqIn7a9kS+juwg=","Payload":{"body":"eyJhcGlWZXJzaW9uIjoiMC4wLjEiLCJraW5kIjoiaGFzaGVkcmVrb3JkIiwic3BlYyI6eyJkYXRhIjp7Imhhc2giOnsiYWxnb3JpdGhtIjoic2hhMjU2IiwidmFsdWUiOiJiNGU3MTNjMjg3YjEyNjczN2IzZWIxY2I4M2NlZWUwMDcwNjAzMjE2YmFiMzI4ODllMDIyNWRjNjEwZjlhNWRkIn19LCJzaWduYXR1cmUiOnsiY29udGVudCI6Ik1FUUNJRi96Y2M0RnNNNDlkc00xYURtUThpZ1Bucmc5SUprZGhVOTV5YzEyaHczVUFpQmUreENmSGZVaTMyc05McWROdzFTbGZJdTZxU29OQjRRWkptdHc4NUpxbWc9PSIsInB1YmxpY0tleSI6eyJjb250ZW50IjoiTFMwdExTMUNSVWRKVGlCRFJWSlVTVVpKUTBGVVJTMHRMUzB0Q2sxSlNVaGlla05EUW5aaFowRjNTVUpCWjBsVlJFWlBZWEpyTVM4cmF6SnBOa3BUTWtaWmRFODFWRWxMU1RZd2QwTm5XVWxMYjFwSmVtb3dSVUYzVFhjS1RucEZWazFDVFVkQk1WVkZRMmhOVFdNeWJHNWpNMUoyWTIxVmRWcEhWakpOVWpSM1NFRlpSRlpSVVVSRmVGWjZZVmRrZW1SSE9YbGFVekZ3WW01U2JBcGpiVEZzV2tkc2FHUkhWWGRJYUdOT1RXcFJkMDlFUlRKTmFrRjNUa1JWZUZkb1kwNU5hbEYzVDBSRk1rMXFRWGhPUkZWNFYycEJRVTFHYTNkRmQxbElDa3R2V2tsNmFqQkRRVkZaU1V0dldrbDZhakJFUVZGalJGRm5RVVZ4VTNseFQxQklSbUpwZDJoWlJVOU9VRGxxZVRKclQzQkVNUzlYVUZwMFlXcGtSV3dLUVVsUFRVc3dWR0ZoTDJJMVVYVnVRM1ZKZFhWUGJqaFBZV2hLZW1nMFJGSTBSMVZIVERKMlRXRndSV1ExUVhwMkt6WlBRMEpvVlhkbloxbFNUVUUwUndwQk1WVmtSSGRGUWk5M1VVVkJkMGxJWjBSQlZFSm5UbFpJVTFWRlJFUkJTMEpuWjNKQ1owVkdRbEZqUkVGNlFXUkNaMDVXU0ZFMFJVWm5VVlZKYVdWdENuQkdiVTVvWTNwQ01sTXZha0pVY1VaaFMxTnZlRTlWZDBoM1dVUldVakJxUWtKbmQwWnZRVlV6T1ZCd2VqRlphMFZhWWpWeFRtcHdTMFpYYVhocE5Ga0tXa1E0ZDJkWlNVZEJNVlZrUlZGRlFpOTNValJOU0dGSFpFZG9NR1JJUW5wUGFUaDJXakpzTUdGSVZtbE1iVTUyWWxNNWVtSlhSbmxrUjA1MlltNVNlUXBaVjA0d1lUSnNNRXd5VG05WlYyeDFZa2RzZFdGNU9IVmFNbXd3WVVoV2FVd3paSFpqYlhSdFlrYzVNMk41T1dsa1YyeHpXa014ZDJSWFNuTmhXRTV2Q2t4dWJIUmlSVUo1V2xkYWVrd3pVbWhhTTAxMlpHcEpkVTFVVVhWT2VURnVZak5LYkdKSFZtaGpNbFo1VEZoU2JHTXpVVFJOUkd0SFEybHpSMEZSVVVJS1p6YzRkMEZSUlVWTE1tZ3daRWhDZWs5cE9IWmtSemx5V2xjMGRWbFhUakJoVnpsMVkzazFibUZZVW05a1Ywb3hZekpXZVZreU9YVmtSMVoxWkVNMWFncGlNakIzUldkWlMwdDNXVUpDUVVkRWRucEJRa0ZuVVVWalNGWjZZVVJCTWtKbmIzSkNaMFZGUVZsUEwwMUJSVVJDUTJkNFQxUk9iVTE2V21sT1JGRjVDbGxxWnpOTlIwa3dXWHBGZUZsNlZtaFpWMDB4V1hwWk1FOUViR3BaTWxWNFQxZEdhazVFYkd4TlF6aEhRMmx6UjBGUlVVSm5OemgzUVZGUlJVbFZTakVLWVZkNGEweERRbFJoVjJSMVNVZEdkVnBEUWxGa1YwcHpZVmhPYjBsRlRtOVpWMngxWWtkc2RXRjZRVzlDWjI5eVFtZEZSVUZaVHk5TlFVVkdRa0p3ZWdwaVYwWjVaRWRPZG1KdVVubFpWMDR3WVRKc01Fd3lUbTlaVjJ4MVlrZHNkV0Y2UVhkQ1oyOXlRbWRGUlVGWlR5OU5RVVZIUWtOS2VWcFhXbnBNTTFKb0Nsb3pUWFprYWtsMVRWUlJkVTU1TVc1aU0wcHNZa2RXYUdNeVZubE1XRkpzWXpOUk5FMUVjMGREYVhOSFFWRlJRbWMzT0hkQlVXZEZURkYzY21GSVVqQUtZMGhOTmt4NU9UQmlNblJzWW1rMWFGa3pVbkJpTWpWNlRHMWtjR1JIYURGWmJsWjZXbGhLYW1JeU5UQmFWelV3VEcxT2RtSlVRMEpvUVZsTFMzZFpRZ3BDUVVkRWRucEJRa05SVWpKRVNGSnZaRWhTZDJONmIzWk1NbVJ3WkVkb01WbHBOV3BpTWpCMll6SXhhR051VW1waU1qVXdZMjFHYW1SSGRIQmtRemxxQ21GSFJuQmliWGh3WW0xemRreHRaSEJrUjJneFdXazVNMkl6U25KYWJYaDJaRE5OZGxsdVZuQmlSMUYwWTBoV2FXSkhiSHBoUXpVMVlsZDRRV050Vm0wS1kzazVNRmxYWkhwTU0xbDVUR3BGTUV4cVkzUmFNamw1V2xkNGJGbFlUbXhqYVRFd1dsaE9NRTlFUVRSQ1oyOXlRbWRGUlVGWlR5OU5RVVZMUWtOdlRRcExSRVUxVFRKWmVrNXRTVEJPUkVwcFQwUmpkMWxxVW1wTlZFWnFUbGRHYUZsNlZtcE9hbEUwVDFkT2FscFVSVFZaVjAwd1QxZFZkMGhSV1V0TGQxbENDa0pCUjBSMmVrRkNRM2RSVUVSQk1XNWhXRkp2WkZkSmRHRkhPWHBrUjFaclRVUXdSME5wYzBkQlVWRkNaemM0ZDBGUmQwVk1kM2QwWVVoU01HTklUVFlLVEhrNWJtRllVbTlrVjBsMVdUSTVkRXd6VG5SWldFb3dXVEk1ZFdSSVNtaFpNMUp5WVZoUmRsa3lhR2hoVnpWellWYzFjazFFWjBkRGFYTkhRVkZSUWdwbk56aDNRVkV3UlV0bmQyOU5WR3Q2V21wTk1sbHFVVEJOYlVrMFRucENhVTVIVFhoTlYwMHhXVmRHYWs1WFRUSk9SR2MxV1RKT2JFMVViR2haZWxFMUNscFVRWGxDWjI5eVFtZEZSVUZaVHk5TlFVVlBRa05SVFVsdVNteGFiazEyWkVkR2JtTjVPVEpOYVRSNFRrTTBNMHhYWkhaamJWWnpXbGRHZWxwWVNYUUtaRWRXZW1SRVozZEhVVmxMUzNkWlFrSkJSMFIyZWtGQ1JIZFJURVJCYTNoTlZFVXdUbFJWTkU1cVkzZE5kMWxMUzNkWlFrSkJSMFIyZWtGQ1JVRlJiQXBFUTA1dlpFaFNkMk42YjNaTU1tUndaRWRvTVZscE5XcGlNakIyWXpJeGFHTnVVbXBpTWpVd1kyMUdhbVJIZEhCa1JFRlpRbWR2Y2tKblJVVkJXVTh2Q2sxQlJWSkNRVzlOUTBSSk1VMVVSWGhOUkUxNVRVbEhSVUpuYjNKQ1owVkZRVmxQTDAxQlJWTkNTRmxOWkVkb01HUklRbnBQYVRoMldqSnNNR0ZJVm1rS1RHMU9kbUpUT1hwaVYwWjVaRWRPZG1KdVVubFpWMDR3WVRKc01Fd3lUbTlaVjJ4MVlrZHNkV0Y1T0hWYU1td3dZVWhXYVV3elpIWmpiWFJ0WWtjNU13cGplVGxwWkZkc2MxcERNWGRrVjBwellWaE9iMHh1YkhSaVJVSjVXbGRhZWt3elVtaGFNMDEyWkdwSmRVMVVVWFZPZVRGdVlqTktiR0pIVm1oak1sWjVDa3hZVW14ak0xRTBUVVJuUjBOcGMwZEJVVkZDWnpjNGQwRlNUVVZMWjNkdlRWUnJlbHBxVFRKWmFsRXdUVzFKTkU1NlFtbE9SMDE0VFZkTk1WbFhSbW9LVGxkTk1rNUVaelZaTWs1c1RWUnNhRmw2VVRWYVZFRlZRbWR2Y2tKblJVVkJXVTh2VFVGRlZVSkJXVTFDU0VJeFl6Sm5kMWxSV1V0TGQxbENRa0ZIUkFwMmVrRkNSbEZTVkVSR1JtOWtTRkozWTNwdmRrd3laSEJrUjJneFdXazFhbUl5TUhaak1qRm9ZMjVTYW1JeU5UQmpiVVpxWkVkMGNHUkRPV3BoUjBad0NtSnRlSEJpYlhOMldWZE9NR0ZYT1hWamVUbDVaRmMxZWt4NlJYZE9SRWt4VGtScmVVOVVWWHBNTWtZd1pFZFdkR05JVW5wTWVrVjNSbWRaUzB0M1dVSUtRa0ZIUkhaNlFVSkdaMUZKUkVGYWQyUlhTbk5oVjAxM1oxbHZSME5wYzBkQlVWRkNNVzVyUTBKQlNVVm1RVkkyUVVoblFXUm5SR1JRVkVKeGVITmpVZ3BOYlUxYVNHaDVXbHA2WTBOdmEzQmxkVTQwT0hKbUswaHBia3RCVEhsdWRXcG5RVUZCV2taamVYWnhjVUZCUVVWQmQwSklUVVZWUTBsQ2RXWkxaMWw0Q21NclYyNUJlV0pTY0dWWUt5dEJSMFJGYlROV1MwNW1XbEo2Y0hsS1RWTkRXRXBYV0VGcFJVRnVWeXQwWW5Oc2FEbDVUbUZ5TW5CbllXVlhUVmN5VEdrS1YxbFhVbWNyY3pOVFlXeFRkMmxVZUZSS2EzZERaMWxKUzI5YVNYcHFNRVZCZDAxRVduZEJkMXBCU1hkRVRVTklaMm92VWtvNFV6TklhWGw2ZG5KM0x3cHNjWEZ3UzNCYVlUTXZRaTlHUlZJMGIwUlpSVTFRT1hCbWJYcHBaMlZuZG5Sc1RGZEJSbUpzUmtjMWJrRnFRamRqY1ZOUVNXcG1ObUp4UVNzdllYQlVDblZaWmxOeVFtSjNValJKVkZNek5qZHhURTVqVVhSWGFuUjZhbkZvV1VWcGVFaFFZMDFaTlcwNFZIQk1XbE5uUFFvdExTMHRMVVZPUkNCRFJWSlVTVVpKUTBGVVJTMHRMUzB0Q2c9PSJ9fX19","integratedTime":1723838692,"logIndex":122141623,"logID":"c0d23d6ad406973f9559f3ba2d1ca01f84147d8ffc5b8445c224f98b9591801d"}},"Issuer":"https://token.actions.githubusercontent.com","Subject":"https://github.com/smartcontractkit/chainlink/.github/workflows/build-publish.yml@refs/tags/v2.14.7-goreleaser-test8","githubWorkflowName":"Build, Sign and Publish Chainlink","githubWorkflowRef":"refs/tags/v2.14.7-goreleaser-test8","githubWorkflowRepository":"smartcontractkit/chainlink","githubWorkflowSha":"193f36b442b870b4c11c5aac5c6489cce19ac49e","githubWorkflowTrigger":"push"}},{"critical":{"identity":{"docker-reference":"public.ecr.aws/chainlink/chainlink"},"image":{"docker-manifest-digest":"sha256:ff509c9118c14d2b45055c9c5b9ed203f931bb07e74ee0317f15c97d808de1cf"},"type":"cosign container image signature"},"optional":{"1.3.6.1.4.1.57264.1.1":"https://token.actions.githubusercontent.com","1.3.6.1.4.1.57264.1.2":"push","1.3.6.1.4.1.57264.1.3":"193f36b442b870b4c11c5aac5c6489cce19ac49e","1.3.6.1.4.1.57264.1.4":"Build, Sign and Publish Chainlink","1.3.6.1.4.1.57264.1.5":"smartcontractkit/chainlink","1.3.6.1.4.1.57264.1.6":"refs/tags/v2.14.7-goreleaser-test8","Bundle":{"SignedEntryTimestamp":"MEQCIA0e/Bh3BRjIlXiweG+7yW9NenluwYcNvll96aAK/zT6AiBV0RRIE/ehJ0y2mt2RblfmEpP8E8fw3yLQAcFoaKs1Rg==","Payload":{"body":"eyJhcGlWZXJzaW9uIjoiMC4wLjEiLCJraW5kIjoiaGFzaGVkcmVrb3JkIiwic3BlYyI6eyJkYXRhIjp7Imhhc2giOnsiYWxnb3JpdGhtIjoic2hhMjU2IiwidmFsdWUiOiJiNGU3MTNjMjg3YjEyNjczN2IzZWIxY2I4M2NlZWUwMDcwNjAzMjE2YmFiMzI4ODllMDIyNWRjNjEwZjlhNWRkIn19LCJzaWduYXR1cmUiOnsiY29udGVudCI6Ik1FUUNJRTBTekd0V3lOL2xTQmo0UGIrSnVBbERaSGs3NEZoRnQ3YSs4VkY0R3JoUkFpQmpmQWdZQTh1cDZRQnE1MW8rd3RpTlRXWVZ2MDYzOGFnVm41SjU3ZE16Q3c9PSIsInB1YmxpY0tleSI6eyJjb250ZW50IjoiTFMwdExTMUNSVWRKVGlCRFJWSlVTVVpKUTBGVVJTMHRMUzB0Q2sxSlNVaGlla05EUW5aaFowRjNTVUpCWjBsVlVISTRlRWN3Wm1OS09YTkxOWEpWUWxaWVJGaGlZV1JDUkhGbmQwTm5XVWxMYjFwSmVtb3dSVUYzVFhjS1RucEZWazFDVFVkQk1WVkZRMmhOVFdNeWJHNWpNMUoyWTIxVmRWcEhWakpOVWpSM1NFRlpSRlpSVVVSRmVGWjZZVmRrZW1SSE9YbGFVekZ3WW01U2JBcGpiVEZzV2tkc2FHUkhWWGRJYUdOT1RXcFJkMDlFUlRKTmFrRjNUa1JWTVZkb1kwNU5hbEYzVDBSRk1rMXFRWGhPUkZVeFYycEJRVTFHYTNkRmQxbElDa3R2V2tsNmFqQkRRVkZaU1V0dldrbDZhakJFUVZGalJGRm5RVVZYZVRSeFNIaGlOUzlYVDJzeVkwVkRXalJTUkdobFFVVmxZakZKZFdoTU9WQm1XVTBLTkU4MksybDROR1owVHpkb1FsbFhNRzRyZW5sclNIcHpkV001UWk5QlNtZzNjVVpUV0VKQ2VEQldOVEJHYWxaNlEwdFBRMEpvVlhkbloxbFNUVUUwUndwQk1WVmtSSGRGUWk5M1VVVkJkMGxJWjBSQlZFSm5UbFpJVTFWRlJFUkJTMEpuWjNKQ1owVkdRbEZqUkVGNlFXUkNaMDVXU0ZFMFJVWm5VVlZSVERWSUNtSm1iMFIyZWxJNWNTOUlaRzR6YWs1TFZHd3hNMlpSZDBoM1dVUldVakJxUWtKbmQwWnZRVlV6T1ZCd2VqRlphMFZhWWpWeFRtcHdTMFpYYVhocE5Ga0tXa1E0ZDJkWlNVZEJNVlZrUlZGRlFpOTNValJOU0dGSFpFZG9NR1JJUW5wUGFUaDJXakpzTUdGSVZtbE1iVTUyWWxNNWVtSlhSbmxrUjA1MlltNVNlUXBaVjA0d1lUSnNNRXd5VG05WlYyeDFZa2RzZFdGNU9IVmFNbXd3WVVoV2FVd3paSFpqYlhSdFlrYzVNMk41T1dsa1YyeHpXa014ZDJSWFNuTmhXRTV2Q2t4dWJIUmlSVUo1V2xkYWVrd3pVbWhhTTAxMlpHcEpkVTFVVVhWT2VURnVZak5LYkdKSFZtaGpNbFo1VEZoU2JHTXpVVFJOUkd0SFEybHpSMEZSVVVJS1p6YzRkMEZSUlVWTE1tZ3daRWhDZWs5cE9IWmtSemx5V2xjMGRWbFhUakJoVnpsMVkzazFibUZZVW05a1Ywb3hZekpXZVZreU9YVmtSMVoxWkVNMWFncGlNakIzUldkWlMwdDNXVUpDUVVkRWRucEJRa0ZuVVVWalNGWjZZVVJCTWtKbmIzSkNaMFZGUVZsUEwwMUJSVVJDUTJkNFQxUk9iVTE2V21sT1JGRjVDbGxxWnpOTlIwa3dXWHBGZUZsNlZtaFpWMDB4V1hwWk1FOUViR3BaTWxWNFQxZEdhazVFYkd4TlF6aEhRMmx6UjBGUlVVSm5OemgzUVZGUlJVbFZTakVLWVZkNGEweERRbFJoVjJSMVNVZEdkVnBEUWxGa1YwcHpZVmhPYjBsRlRtOVpWMngxWWtkc2RXRjZRVzlDWjI5eVFtZEZSVUZaVHk5TlFVVkdRa0p3ZWdwaVYwWjVaRWRPZG1KdVVubFpWMDR3WVRKc01Fd3lUbTlaVjJ4MVlrZHNkV0Y2UVhkQ1oyOXlRbWRGUlVGWlR5OU5RVVZIUWtOS2VWcFhXbnBNTTFKb0Nsb3pUWFprYWtsMVRWUlJkVTU1TVc1aU0wcHNZa2RXYUdNeVZubE1XRkpzWXpOUk5FMUVjMGREYVhOSFFWRlJRbWMzT0hkQlVXZEZURkYzY21GSVVqQUtZMGhOTmt4NU9UQmlNblJzWW1rMWFGa3pVbkJpTWpWNlRHMWtjR1JIYURGWmJsWjZXbGhLYW1JeU5UQmFWelV3VEcxT2RtSlVRMEpvUVZsTFMzZFpRZ3BDUVVkRWRucEJRa05SVWpKRVNGSnZaRWhTZDJONmIzWk1NbVJ3WkVkb01WbHBOV3BpTWpCMll6SXhhR051VW1waU1qVXdZMjFHYW1SSGRIQmtRemxxQ21GSFJuQmliWGh3WW0xemRreHRaSEJrUjJneFdXazVNMkl6U25KYWJYaDJaRE5OZGxsdVZuQmlSMUYwWTBoV2FXSkhiSHBoUXpVMVlsZDRRV050Vm0wS1kzazVNRmxYWkhwTU0xbDVUR3BGTUV4cVkzUmFNamw1V2xkNGJGbFlUbXhqYVRFd1dsaE9NRTlFUVRSQ1oyOXlRbWRGUlVGWlR5OU5RVVZMUWtOdlRRcExSRVUxVFRKWmVrNXRTVEJPUkVwcFQwUmpkMWxxVW1wTlZFWnFUbGRHYUZsNlZtcE9hbEUwVDFkT2FscFVSVFZaVjAwd1QxZFZkMGhSV1V0TGQxbENDa0pCUjBSMmVrRkNRM2RSVUVSQk1XNWhXRkp2WkZkSmRHRkhPWHBrUjFaclRVUXdSME5wYzBkQlVWRkNaemM0ZDBGUmQwVk1kM2QwWVVoU01HTklUVFlLVEhrNWJtRllVbTlrVjBsMVdUSTVkRXd6VG5SWldFb3dXVEk1ZFdSSVNtaFpNMUp5WVZoUmRsa3lhR2hoVnpWellWYzFjazFFWjBkRGFYTkhRVkZSUWdwbk56aDNRVkV3UlV0bmQyOU5WR3Q2V21wTk1sbHFVVEJOYlVrMFRucENhVTVIVFhoTlYwMHhXVmRHYWs1WFRUSk9SR2MxV1RKT2JFMVViR2haZWxFMUNscFVRWGxDWjI5eVFtZEZSVUZaVHk5TlFVVlBRa05SVFVsdVNteGFiazEyWkVkR2JtTjVPVEpOYVRSNFRrTTBNMHhYWkhaamJWWnpXbGRHZWxwWVNYUUtaRWRXZW1SRVozZEhVVmxMUzNkWlFrSkJSMFIyZWtGQ1JIZFJURVJCYTNoTlZFVXdUbFJWTkU1cVkzZE5kMWxMUzNkWlFrSkJSMFIyZWtGQ1JVRlJiQXBFUTA1dlpFaFNkMk42YjNaTU1tUndaRWRvTVZscE5XcGlNakIyWXpJeGFHTnVVbXBpTWpVd1kyMUdhbVJIZEhCa1JFRlpRbWR2Y2tKblJVVkJXVTh2Q2sxQlJWSkNRVzlOUTBSSk1VMVVSWGhOUkUxNVRVbEhSVUpuYjNKQ1owVkZRVmxQTDAxQlJWTkNTRmxOWkVkb01HUklRbnBQYVRoMldqSnNNR0ZJVm1rS1RHMU9kbUpUT1hwaVYwWjVaRWRPZG1KdVVubFpWMDR3WVRKc01Fd3lUbTlaVjJ4MVlrZHNkV0Y1T0hWYU1td3dZVWhXYVV3elpIWmpiWFJ0WWtjNU13cGplVGxwWkZkc2MxcERNWGRrVjBwellWaE9iMHh1YkhSaVJVSjVXbGRhZWt3elVtaGFNMDEyWkdwSmRVMVVVWFZPZVRGdVlqTktiR0pIVm1oak1sWjVDa3hZVW14ak0xRTBUVVJuUjBOcGMwZEJVVkZDWnpjNGQwRlNUVVZMWjNkdlRWUnJlbHBxVFRKWmFsRXdUVzFKTkU1NlFtbE9SMDE0VFZkTk1WbFhSbW9LVGxkTk1rNUVaelZaTWs1c1RWUnNhRmw2VVRWYVZFRlZRbWR2Y2tKblJVVkJXVTh2VFVGRlZVSkJXVTFDU0VJeFl6Sm5kMWxSV1V0TGQxbENRa0ZIUkFwMmVrRkNSbEZTVkVSR1JtOWtTRkozWTNwdmRrd3laSEJrUjJneFdXazFhbUl5TUhaak1qRm9ZMjVTYW1JeU5UQmpiVVpxWkVkMGNHUkRPV3BoUjBad0NtSnRlSEJpYlhOMldWZE9NR0ZYT1hWamVUbDVaRmMxZWt4NlJYZE9SRWt4VGtScmVVOVVWWHBNTWtZd1pFZFdkR05JVW5wTWVrVjNSbWRaUzB0M1dVSUtRa0ZIUkhaNlFVSkdaMUZKUkVGYWQyUlhTbk5oVjAxM1oxbHZSME5wYzBkQlVWRkNNVzVyUTBKQlNVVm1RVkkyUVVoblFXUm5SR1JRVkVKeGVITmpVZ3BOYlUxYVNHaDVXbHA2WTBOdmEzQmxkVTQwT0hKbUswaHBia3RCVEhsdWRXcG5RVUZCV2taamVYZHRWRUZCUVVWQmQwSklUVVZWUTBsRGFqVkNTblZIQ2k4M1ZHeHFNbkkyYlhJMVRVWnVVbWhPVjB4VVdrSlZRVFZWVm5BeWNYWkJSMFJPY0VGcFJVRXdTazl5Vmtvd1NYSXJRbVZoWml0NVYxVkZWMDh3VldvS09XeDNiWEZtVURJMGVGVnFhVGRwWXpreVJYZERaMWxKUzI5YVNYcHFNRVZCZDAxRVduZEJkMXBCU1hkVFJHVmlVR3BWZVc5VmFtOUpWRVpxWjFjNWNRcGpPVVpaTDJsUlFrOUVXbFZNY0U5cGFITjNUMGxyVjJ0dFZYbERhV05OWmk5d1MzWnBZMGhuU2xnMGFVRnFRbXhvVkZGYWRIaDRTakl3TjFOdFRESnhDbEJyYjIxdE5EUjBZVTFWU2xWdGNEaERablZaYXpKQlpWWXZUa0pNTVZSTlNWa3lLM2xWVUZSc05tRjZaalF3UFFvdExTMHRMVVZPUkNCRFJWSlVTVVpKUTBGVVJTMHRMUzB0Q2c9PSJ9fX19","integratedTime":1723838696,"logIndex":122141632,"logID":"c0d23d6ad406973f9559f3ba2d1ca01f84147d8ffc5b8445c224f98b9591801d"}},"Issuer":"https://token.actions.githubusercontent.com","Subject":"https://github.com/smartcontractkit/chainlink/.github/workflows/build-publish.yml@refs/tags/v2.14.7-goreleaser-test8","githubWorkflowName":"Build, Sign and Publish Chainlink","githubWorkflowRef":"refs/tags/v2.14.7-goreleaser-test8","githubWorkflowRepository":"smartcontractkit/chainlink","githubWorkflowSha":"193f36b442b870b4c11c5aac5c6489cce19ac49e","githubWorkflowTrigger":"push"}}]
```

### for regular docker image build:

https://github.com/smartcontractkit/chainlink/actions/runs/10458852645/job/28961572519

After refactor based on suggestions:
https://github.com/smartcontractkit/chainlink/actions/runs/10475918902/job/29013711522

## TODO

After this PR is merged, remove these secrets from repo as we have no use for them using OIDC keyless signing:

- `COSIGN_PASSWORD`
- `COSIGN_PRIVATE_KEY`
- `COSIGN_PUBLIC_KEY`

### CODE REVIEW

**Review by erikburt — COMMENTED** [review]
How does keyless signing change the process for verifying the authenticity of the signed images? How is this update being communicated to people who do this (ie. NOPs)?

File: .github/actions/build-sign-publish-chainlink/action.yml [hunk 1]
```diff
@@ -239,74 +230,34 @@ runs:
...
-    - if: inputs.verify-signature == 'true' && inputs.sign-method == 'keypair'
-      name: Verify the signature of the published non-root Docker image using keypair
-      shell: sh
       run: |
-        echo "${{ inputs.cosign-public-key }}" > cosign.key
-        cosign verify --key cosign.key "${{ env.nonroot_image_name }}"
-        rm -f cosign.key
+        cosign verify "${{ env.root_image_name }}" \
+        --certificate-oidc-issuer https://token.actions.githubusercontent.com \
+        --certificate-identity "https://github.com/smartcontractkit/chainlink/.github/workflows/build-publish.yml@${{ github.ref }}"
```
Comment (HenryNguyen5) [inline .github/actions/build-sign-publish-chainlink/action.yml]: Should we share vars here since we use it twice?

**Review by HenryNguyen5 — COMMENTED** [review]
this is awesome, is there a reason why we wouldnt use https://docs.github.com/en/actions/security-for-github-actions/using-artifact-attestations/using-artifact-attestations-to-establish-provenance-for-builds#what-to-attest
cc @javuto

**Comment by momentmaker** [comment]
> How does keyless signing change the process for verifying the authenticity of the signed images? How is this update being communicated to people who do this (ie. NOPs)?

yeah that's a good point - I don't think they're currently verifying the currently built images. right @chainchad ?

we can make it into the release notes for the next version as well as make an announcement in the respective channels?

**Comment by chainchad** [comment]
> yeah that's a good point - I don't think they're currently verifying the currently built images. right @chainchad ?

AFAIK: Not outside of verifying the sha digest.

@momentmaker yeah, we'll want to have some public documentation for how this should be verified and potentially add it to our main docs page cc @dwightjl.

File: .github/actions/build-sign-publish-chainlink/action.yml [hunk 2]
```diff
@@ -51,23 +51,11 @@ inputs:
     description: When set to the string boolean value of "true", the resulting build image will be signed
     default: "false"
     required: false
-  cosign-private-key:
```
Comment (chainchad) [inline .github/actions/build-sign-publish-chainlink/action.yml:54]: If we go with this, we should remove the current secret on the repo.

cc @javuto

File: .github/actions/build-sign-publish-chainlink/action.yml [hunk 3]
```diff
@@ -239,74 +230,34 @@ runs:
...
 
-    - if: inputs.sign-images == 'true' && inputs.sign-method == 'keyless'
+    - if: inputs.sign-images == 'true'
       name: Sign the published root Docker image using keyless method
       shell: sh
-      env:
-        COSIGN_EXPERIMENTAL: 1
       run: |
-        cosign sign "${{ env.root_image_name }}"
+        cosign sign "${{ env.root_image_name }}" --yes
```
Comment (chainchad) [inline .github/actions/build-sign-publish-chainlink/action.yml:247]: Will this automatically sign with the correct OIDC provider from GH? If so, maybe we should add a comment.

File: .github/actions/build-sign-publish-chainlink/action.yml (hunk 2 above)
Comment (momentmaker) [reply .github/actions/build-sign-publish-chainlink/action.yml:54]: I'll add to the description for TODO

**Review by chainchad — APPROVED** [review]
👍 👍 

Would want @smartcontractkit/prodsec-public input too before merging.

**Comment by momentmaker** [comment]
> this is awesome, is there a reason why we wouldnt use [docs.github.com/en/actions/security-for-github-actions/using-artifact-attestations/using-artifact-attestations-to-establish-provenance-for-builds#what-to-attest](https://docs.github.com/en/actions/security-for-github-actions/using-artifact-attestations/using-artifact-attestations-to-establish-provenance-for-builds#what-to-attest) cc @javuto

yeah I'm actually gonna work on this in another ticket - gonna be something similar to this:
https://github.com/goreleaser/goreleaser/issues/4852

File: .github/actions/build-sign-publish-chainlink/action.yml [hunk 4]
```diff
@@ -84,6 +72,8 @@ runs:
       # See https://docs.github.com/en/actions/learn-github-actions/workflow-commands-for-github-actions#multiline-strings
       run: |
         SHARED_IMAGES=${{ inputs.ecr-hostname }}/${{ inputs.ecr-image-name }}
+        OIDC_ISSUER=https://token.actions.githubusercontent.com
+        OIDC_IDENTITY=https://github.com/smartcontractkit/chainlink/.github/workflows/build-publish.yml@${{ github.ref }}
```
Comment (erikburt) [inline .github/actions/build-sign-publish-chainlink/action.yml:76]: Is this technically forgeable? Like I could update this action in a PR and get an image signed for any version by hardcoding this?

**Review by erikburt — APPROVED** [review]

**Review by javuto — COMMENTED** [review]
Nice job!

**Pull request merged**